storm-core/src/clj/org/apache/storm/daemon/common.clj

```diff
@@ -298,18 +299,21 @@
                       {[comp-id METRICS-STREAM-ID] :shuffle})
                     (into {}))
         
-        mk-bolt-spec (fn [class arg p]
+        mk-bolt-spec (fn [class arg p max-retain-metric-tuples whitelist blacklist]
                        (thrift/mk-bolt-spec*
                         inputs
-                        (org.apache.storm.metric.MetricsConsumerBolt. class arg)
+                        (org.apache.storm.metric.MetricsConsumerBolt. class arg max-retain-metric-tuples (FilterByMetricName. whitelist blacklist))
```

Comment:
Are MetricsFilters intended to be pluggable, or do we only want to allow filtering by name?